src/rgw/rgw_formats.cc

```diff
@@ -248,7 +248,7 @@ void RGWFormatter_Plain::write_data(const char *fmt, ...)
     free(p);
 }
 
-void RGWFormatter_Plain::dump_value_int(const char *name, const char *fmt, ...)
+void RGWFormatter_Plain::dump_value_int(std::string_view name, const char *fmt, ...)
```

Comment:
seeing a regression in rgw here. `name` is being used below in `write_data("%s%s: %s", eol, name, buf);`. since this is fed into printf-style varargs, the compiler didn't catch this:
```
*** Caught signal (Segmentation fault) **
 in thread 7f153bff7700 thread_name:radosgw
 ceph version 15.1.0-1576-gb5c14b8 (b5c14b87315b02feb7252b55cb060e536195651d) octopus (rc)
 1: (()+0x3ef20) [0x7f1645b9bf20]
 2: (()+0x18e5a1) [0x7f1645ceb5a1]
 3: (_IO_vfprintf()+0x2143) [0x7f1645bba4d3]
 4: (__vsnprintf_chk()+0xa9) [0x7f1645c8f169]
 5: (RGWFormatter_Plain::write_data(char const*, ...)+0x102) [0x7f16464877d2]
 6: (RGWFormatter_Plain::dump_value_int(std::basic_string_view<char, std::char_traits<char> >, char const*, ...)+0x192) [0x7f1646487c92]
 7: (RGWBulkUploadOp_ObjStore_SWIFT::send_response()+0x163) [0x7f1646343da3]
```
fixed this and a related issue in https://github.com/ceph/ceph/pull/33754

Reply:
@cbodley thanks for the fix and sorry for the regression =(